### PR TITLE
Rewrite spec.md for 1.0.0

### PR DESCRIPTION
**Pull request by squeed**
The protocol is unchanged, but the specification was badly in need of a rewrite.

### CODE REVIEW

File: SPEC.md
Comment (bboreham) [inline SPEC.md:66]: I would say "semver" is more commonly understood than "Semantic Version".

File: SPEC.md
Comment (bboreham) [inline SPEC.md:67]: s/a alphanumeric/an alphanumeric/

File: SPEC.md
Comment (bboreham) [inline SPEC.md:30]: I think we should be explicit that non-containers such as VMs are allowed, and the fields are just re-purposed

File: SPEC.md
Comment (bboreham) [inline SPEC.md:291]: What is this line doing?

File: SPEC.md
Comment (bboreham) [inline SPEC.md:315]: s/attach create/attach/

**Comment by squeed** [comment]
CNI maintainers: any more changes?

File: SPEC.md
Comment (dcbw) [inline SPEC.md:67]: @squeed "across all containers"?

**Review by bboreham — APPROVED** [review]
Excellent work!  Let's merge this and clean up anything else after.

**Review by mars1024 — APPROVED** [review]
/lgtm, a huge and professional job!

File: SPEC.md
Comment (s1061123) [inline SPEC.md:372]: If `args` and `CNI_ARGS` gives different parameters on same argument, what is happen? Is that in scope of SPEC?

```
CNI_ARGS=argA=bar

{
    "cniVersion": "1.0.0",
    "name": "dbnet",
    "type": "bridge",
    "bridge": "cni0",
    "keyA": ["some more", "plugin specific", "configuration"],
    "args": {"argA": "foo"},
    "ipam": {
        "type": "host-local",
        "subnet": "10.1.0.0/16",
        "gateway": "10.1.0.1"
    },
    "dns": {
        "nameservers": [ "10.1.0.1" ]
    }
}
```

**Comment by passcod** [comment]
Shouldn't the table of contents be re-added?

**Comment by squeed** [comment]
Good point, added a TOC.

**Pull request merged**
